internal/utils/akka/src/test/java/org/eclipse/ditto/internal/utils/akka/controlflow/TimeMeasuringFlowTest.java

```diff
@@ -129,71 +134,112 @@ public void timeMeasuresOnlyTimeOfWrappedFlow() {
         }};
     }
 
     @Test
-    @Ignore
-    public void keepsParallelism() {
-        final Duration sleepDuration = Duration.ofMillis(100);
-        final Flow<String, String, NotUsed> flowThatNeedsSomeTimeButUsesParallelism =
-                Flow.<String>create().flatMapMerge(10, input -> Source.single(input).via
-                        (Flow.<String, String>fromFunction(x -> {
-                            TimeUnit.MILLISECONDS.sleep(sleepDuration.toMillis());
-                            return x;
-                        }).async()));
+    public void keepsParallelism() throws Exception {
+        final Duration sleepDuration = Duration.ofMillis(1000);
+        final Flow<String, String, NotUsed> flowThatNeedsSomeTimeButUsesParallelism = Flow.<String>create()
+                .flatMapMerge(1000, input -> Source.single(input)
+                        .delay(sleepDuration, DelayOverflowStrategy.backpressure()));
 
         final PreparedTimer timer = DittoMetrics.timer("test-time-measuring-flow");
         final PreparedTimer timerMock = mock(PreparedTimer.class);
-        final int numberOfRepetitions = 10;
+        final int numberOfRepetitions = 160;
+        final Queue<Duration> durations = new ConcurrentLinkedQueue<>();
         when(timerMock.start()).thenAnswer(AdditionalAnswers.delegatesTo(timer));
         final List<String> expectedResults = IntStream.range(0, numberOfRepetitions)
                 .mapToObj(i -> "Test")
                 .collect(Collectors.toList());
         new TestKit(system) {{
             Source.from(expectedResults)
-                    .via(TimeMeasuringFlow.measureTimeOf(flowThatNeedsSomeTimeButUsesParallelism, timerMock))
+                    .via(TimeMeasuringFlow.measureTimeOf(flowThatNeedsSomeTimeButUsesParallelism, timerMock, 170,
+                            i -> {}, durations::add))
                     .to(testSink)
                     .run(system);
             final Instant start = Instant.now();
             sinkProbe.request(numberOfRepetitions);
             sinkProbe.expectNextN(CollectionConverters.asScala(expectedResults).toSeq());
             final Instant end = Instant.now();
             final Duration duration = Duration.ofMillis(end.toEpochMilli() - start.toEpochMilli());
-            assertThat(duration).isLessThan(sleepDuration.multipliedBy(numberOfRepetitions));
+            assertThat(duration).isLessThan(sleepDuration.multipliedBy(3));
             verify(timerMock, times(numberOfRepetitions)).start();
+
+            if (durations.size() < numberOfRepetitions) {
+                Thread.sleep(sleepDuration.toMillis());
+            }
+
+            final double averageDurationInNanos = durations.stream()
+                    .mapToLong(Duration::toNanos)
+                    .average()
+                    .orElseThrow();
+            // allow 50% inaccuracy due to load
```

Comment:
See question above. If the measured time is that inaccurate, I think it is no longer valuable.

Reply:
Yeah maybe it is better to remove the component. A 250ms -500ms scheduling delay seems unavoidable.